### ISSUE TITLE
Bson unable to find codec

### ISSUE THREAD

**Issue by DeonBoon** [quarkusio/quarkus#7802]
Hi all, I am using Quarkus to build a web service which takes in an input, processes it and stores some results as objects into MongoDB. However, it is throwing me an error about "Bson unable to find a codec". I believe it should be related to how my object is constructed.

Sample attributes of my object:
```
public class ResultsResponse  {
    //Job id for tracking if using db to store
    private String jobId;
    private Boolean matched;
    private long timeElapsedInMillis;
    private String source1;
    private String source2;
    private Double absoluteTolerance;
    private Integer maxResultsLimit;
    private List<List<String>> missingRows;
    private List<String> headers;
    private List<List<String>> acceptableRows;
} 
```
From the googling that I have done, I believe it is because Bson is unable to map a List<List<String> into a mongodb data type. I also followed the quarkus documentation [here](https://quarkus.io/guides/mongodb#configuring-the-mongodb-database) to create a custom codec provider but I don't quite understand how exactly I can do it.

Appreciate if anyone is able to provide any inputs and feel free to ask for more information if needed. Thank you!

**Comment by geoand** [comment]
cc @loicmathieu

**Comment by loicmathieu** [comment]
@DeonBoon can you give us the stacktrace ?
Do you use MongoDB with Panache or directly the MongoDB client ?

**Comment by DeonBoon** [comment]
@loicmathieu the stacktrace only appeared once and I did not manage to capture it. Subsequently the stacktrace did not appear and my application just hangs there after I started it. Could you perhaps share if there are other methods of capturing the stacktrace?

The applications hangs there only after I imported the panache dependency and added the PanacheMongoEntity to my object class. Without the dependency, my application is able to work with the correct output produced. I am also using Apache Spark in my application if this info helps.

**Comment by DeonBoon** [comment]
@loicmathieu just an update, i created my custom codec and also added "quarkus.log.level=ERROR" in my application.properties and now my application works as intended. However if I were to remove "quarkus.log.level=ERROR", it goes back to hanging without any error or stacktrace.

**Comment by loicmathieu** [comment]
@DeonBoon juste for the record can you give us your application.properties and the codec you created. 

I really don't understand what's going on with your stacktrace (but it's a separate issue), do you use the Quarkus logging support or not ?
